R/parser.R

```diff
@@ -302,7 +302,37 @@ partition_chunk = function(engine, code) {
   meta = substr(src, nchar(s1) + 1, nchar(src) - nchar(s2))
   # see if the metadata looks like YAML or CSV
   if (grepl('^[^ :]+:($|\\s)', meta[1])) {
-    meta = yaml::yaml.load(meta, handlers = list(expr = parse_only))
+    meta = tryCatch(
+      yaml::yaml.load(meta, handlers = list(expr = parse_only)),
+      error = function(e) {
+        message = e$message
+        regex = "line (?<line>\\d+), column (?<column>\\d+)"
+        regex_result = regexpr(regex, message, perl = TRUE)
+        starts = attr(regex_result, "capture.start")
+        lengths = attr(regex_result, "capture.length")
+
+        line_index = substr(message, starts[,"line"], starts[,"line"] + lengths[,"line"] - 1)
+        column_index = substr(message, starts[,"column"], starts[,"column"] + lengths[,"column"] - 1)
+
+        line_index = as.integer(line_index)
+        column_index = as.integer(column_index)
```

Comment:
You can use `regmatches()` with `regexec()`:

```suggestion
        x = e$message
        r = "line (?<row>\\d+), column (?<col>\\d+)"
        m = regmatches(x, regexec(r, x, perl = TRUE))[[1]][-1]
        row = as.integer(m['row'])
        col = as.integer(m['col'])
```